qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -599,29 +572,23 @@ def _clean_validate_template(cls, md_template, study_id, obj,
             raise QiitaDBDuplicateHeaderError(
                 find_duplicates(md_template.columns))
 
-        # We need to check for some special columns, that are not present on
-        # the database, but depending on the data type are required.
-        missing = cls._check_special_columns(md_template, obj)
+        # Check if we have the columns required for some functionality
+        warning_msg = []
+        for key, restriction in viewitems(restriction_dict):
+            missing = set(restriction.columns).difference(md_template)
 
-        conn_handler = conn_handler if conn_handler else SQLConnectionHandler()
-
-        # Get the required columns from the DB
-        db_cols = get_table_cols(cls._table, conn_handler)
-
-        # Remove the sample_id and study_id columns
-        db_cols.remove('sample_id')
-        db_cols.remove(cls._id_column)
+            if missing:
+                warning_msg.append(
+                    "%s: %s" % (restriction.error_msg, ', '.join(missing)))
 
-        # Retrieve the headers of the metadata template
-        headers = list(md_template.keys())
+        if warning_msg:
+            warnings.warn(
+                "Some functionality will be disabled due to missing "
+                "columns:\n\t%s.\nCheck https://github.com/biocore/qiita/wiki"
```

Comment:
First of all, thanks for the compliment, but I am no such thing :P That
being said, this is code that's specifically part of the Model, not of
the View. The View should be concerned with the most convenient
rendering of URLs and in that spirit, we should probably have JS code to
recognize and do the appropriate formatting of URLs. Does that make
sense to everyone?

On (Apr-27-15|16:20), josenavas wrote:
> >  
> > -        # Retrieve the headers of the metadata template
> > -        headers = list(md_template.keys())
> > +        if warning_msg:
> > +            warnings.warn(
> > +                "Some functionality will be disabled due to missing "
> > +                "columns:\n\t%s.\nCheck https://github.com/biocore/qiita/wiki"
> 
> @ElDeveloper you're the CLI guru, what do you think? I prefer not to add HTML code in the CLI error/warning output, but if more people agrees I'm happy to add it...
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1075/files#r29201881